src/realm/cluster_tree.cpp

```diff
@@ -1027,6 +1118,172 @@ void ClusterTree::nullify_links(ObjKey obj_key, CascadeState& state)
+            // Prevent making changes to table that is going to be removed anyway
+            // Furthermore it is a prerequisite for using 'traverse' that the tree
+            // is not modified
+            if (get_owning_table()->links_to_self(col_key)) {
+                return false;
+            }
+            auto col_type = col_key.get_type();
+            if (col_key.is_list() || col_key.is_set()) {
+                if (col_type == col_type_LinkList)
+                    col_type = col_type_Link;
+                if (col_type == col_type_Link) {
+                    ArrayInteger values(alloc);
+                    cluster->init_leaf(col_key, &values);
+                    size_t sz = values.size();
+                    BPlusTree<ObjKey> links(alloc);
+                    for (size_t i = 0; i < sz; i++) {
+                        if (ref_type ref = values.get_as_ref(i)) {
+                            links.init_from_ref(ref);
+                            if (links.size() > 0) {
+                                cluster->remove_backlinks(cluster->get_real_key(i), col_key, links.get_all(), state);
+                            }
+                        }
+                    }
+                }
+                else if (col_type == col_type_TypedLink) {
+                    ArrayInteger values(alloc);
+                    cluster->init_leaf(col_key, &values);
+                    size_t sz = values.size();
+                    BPlusTree<ObjLink> links(alloc);
+                    for (size_t i = 0; i < sz; i++) {
+                        if (ref_type ref = values.get_as_ref(i)) {
+                            links.init_from_ref(ref);
+                            if (links.size() > 0) {
+                                cluster->remove_backlinks(cluster->get_real_key(i), col_key, links.get_all(), state);
+                            }
+                        }
+                    }
+                }
+                else if (col_type == col_type_Mixed) {
+                    ArrayInteger values(alloc);
+                    cluster->init_leaf(col_key, &values);
+                    size_t sz = values.size();
+                    BPlusTree<Mixed> mix_arr(alloc);
+                    for (size_t i = 0; i < sz; i++) {
+                        if (ref_type ref = values.get_as_ref(i)) {
+                            mix_arr.init_from_ref(ref);
+                            auto sz = mix_arr.size();
+                            std::vector<ObjLink> links;
+                            for (size_t j = 0; j < sz; j++) {
+                                auto mix = mix_arr.get(j);
+                                if (mix.is_type(type_TypedLink)) {
+                                    links.push_back(mix.get<ObjLink>());
+                                }
+                            }
+                            if (links.size())
+                                cluster->remove_backlinks(cluster->get_real_key(i), col_key, links, state);
```

Comment:
Good question. I will try to find an answer

Reply:
@ironage  isn't this piece of code a good candidate to be converted using the new class for links introduced here(https://github.com/realm/realm-core/pull/5796#pullrequestreview-1104305243)?

Reply:
Yeah good eye, it is a similar pattern. However, it uses lower level constructs at the cluster level rather than our container accessors. I think that is for performance reasons, so I didn't convert it.